spec/models/user_spec.rb

```diff
@@ -8,4 +8,38 @@
         it { should have_many(:created_reports).class_name('Report').inverse_of(:user) }
         it { should have_many(:created_documents).class_name('Document').inverse_of(:user) }
     end
-  end
\ No newline at end of file
+
+    describe "#user_projects" do
+        let!(:user) { create(:user) }
+        let!(:created_project) { create(:project, user:) }
+
+        context 'when user has no shared projects' do
+            it "returns only the user's created projects" do
+                expect(user.user_projects).to eq([created_project])
+            end
+        end
+
+        context 'when user has shared projects' do
+            let(:shared_project) { create(:project) }
+            let(:shared_resource) { create(:shared_resource, shareable: shared_project, user: user) }
+
+            before { shared_resource }
```

Comment:
```suggestion
            let!(:shared_resource) { create(:shared_resource, shareable: shared_project, user: user) }
```

`let!` will init the factory before the test here.

Reply:
Thanks for catching that, fixed.